### DISCUSSION TITLE
Compiling Static Library with Embarcadero and Clang: Linking Error.

### DISCUSSION THREAD

**Discussion by sashok74** [curl/curl#12450] [Q&A] [answered]
Hello! I'm using Embarcadero with clang. I attempted to compile a static library from the source code, and everything compiles fine without errors. However, during the linking stage when linking the application with this library, a linking error occurs: "Unresolved external '_curl_global_init' referenced from CURL_TEST1.OBJ." I couldn't find information on whether it's possible to compile the library in this way. My own attempts didn't lead to any solution. Now I'm seeking help to understand how to solve my task. Thank you!

**Comment by sashok74** [comment]
Perhaps it will be useful.
This set of definitions solved my problem.
NDEBUG;WIN32;_WINDOWS;_USRDLL;CURL_DISABLE_DOH;CURL_WIN32;CURL_STATICLIB;BUILDING_LIBCURL;CURL_DISABLE_LDAP;USE_WIN32_CRYPTO

**Comment by jay** [comment, marked as answer]
You need to define CURL_STATICLIB when compiling your code, as [mentioned](https://curl.se/docs/faq.html#Link_errors_when_building_libcur) in the FAQ. Other macros like BUILDING_LIBCURL are used only for building libcurl and you shouldn't define them.
